app/Models/Store/OrderItem.php

```diff
@@ -49,6 +58,20 @@ public function refreshCost()
 
     public function getDisplayName()
     {
-        return $this->product->name.($this->extra_info !== null ? " ({$this->extra_info})" : '');
+        if (is_array($this->extra_data)) {
+            $extra_text = '';
+            // this prevents fires if type is not set.
+            if (isset($this->extra_data['type'])) {
+                switch ($this->extra_data['type']) {
```

Comment:
there was a reason earlier for encoding the type into the extra data separately, but eeehhhh..